codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HeaderField.java

```diff
@@ -77,15 +72,15 @@ public int compareTo(HeaderField anotherHeaderField) {
         return ret;
     }
 
-    private int compareTo(byte[] s1, byte[] s2) {
-        int len1 = s1.length;
-        int len2 = s2.length;
+    private static int compareTo(CharSequence s1, CharSequence s2) {
+        int len1 = s1.length();
+        int len2 = s2.length();
         int lim = Math.min(len1, len2);
 
         int k = 0;
         while (k < lim) {
-            byte b1 = s1[k];
-            byte b2 = s2[k];
+            byte b1 = (byte) s1.charAt(k);
```

Comment:
@carl-mastrangelo we use `AsciiString` when adding the headers, so it should be safe. Or I am missing something ?